entity-registry/src/main/java/com/linkedin/metadata/models/registry/ConfigEntityRegistry.java

```diff
@@ -102,16 +102,17 @@ public ConfigEntityRegistry(DataSchemaFactory dataSchemaFactory, InputStream con
     }
     EntitySpecBuilder entitySpecBuilder = new EntitySpecBuilder();
     for (Entity entity : entities.getEntities()) {
-      Optional<DataSchema> entitySchema = dataSchemaFactory.getEntitySchema(entity.getName());
-      if (!entitySchema.isPresent()) {
-        throw new IllegalArgumentException(String.format("Entity %s does not exist", entity.getName()));
-      }
-
       List<AspectSpec> aspectSpecs = new ArrayList<>();
       aspectSpecs.add(getAspectSpec(entity.getKeyAspect(), entitySpecBuilder));
       entity.getAspects().forEach(aspect -> aspectSpecs.add(getAspectSpec(aspect, entitySpecBuilder)));
 
-      EntitySpec entitySpec = entitySpecBuilder.buildEntitySpec(entitySchema.get(), aspectSpecs);
+      EntitySpec entitySpec;
+      Optional<DataSchema> entitySchema = dataSchemaFactory.getEntitySchema(entity.getName());
+      if (!entitySchema.isPresent()) {
+        entitySpec = entitySpecBuilder.buildConfigEntitySpec(entity.getName(), entity.getKeyAspect(), aspectSpecs);
+      } else {
+        entitySpec = entitySpecBuilder.buildEntitySpec(entitySchema.get(), aspectSpecs);
```

Comment:
Why is this necessary? Shouldn't config entity registry always build from config entity spec? Or is this to deal with the possibility that we've already found a snapshot?

Reply:
I think the thought was to create the full entity spec if we have the schema for it. But I don't think it is necessary. Mostly backward compatibility